chainer/link.py

```diff
@@ -552,35 +646,57 @@ def __getitem__(self, name):
         """Equivalent to getattr."""
         return getattr(self, name)
 
+    def __setattr__(self, name, value):
```

Comment:
It is intentional, but I forgot about that till now. I will add this point to the upgrade guide.